subprojects/docs/src/snippets/java-library/module/groovy/build.gradle

```diff
@@ -0,0 +1,37 @@
+plugins {
+    id 'java-library'
+}
+
+repositories {
+    mavenCentral()
+}
+
+// tag::declareVersion[]
+version = '1.1'
+
+tasks.compileJava {
+    options.javaModuleVersion = '1.2'
```

Comment:
Night be good to have a comment, here or in the related documentation (with a callout) to explain that this takes precedence over the project.version